src/AzureExtensionServer/Program.cs

```diff
@@ -114,7 +123,10 @@ private static void HandleCOMServerActivation()
         // This could be called by either of the COM registrations, we will do them all to avoid deadlock and bind all on the extension's lifetime.
         using var extensionServer = new Microsoft.Windows.DevHome.SDK.ExtensionServer();
         var extensionDisposedEvent = new ManualResetEvent(false);
-        var extensionInstance = new AzureExtension(extensionDisposedEvent);
+
+        // Create host with dependency injection
+        using var host = CreateDIHost();

```

Comment:
Just CreateHost? "D" and "I" is hard to read, especially knowing that it returns IHost. There is no need to put design pattern abbreviation into the name.